comp/api/api/apiimpl/internal/agent/agent.go

```diff
@@ -401,23 +405,17 @@ func getCSRFToken(w http.ResponseWriter, _ *http.Request) {
 	w.Write([]byte(gui.CsrfToken))
 }
 
-func getConfigCheck(w http.ResponseWriter, _ *http.Request) {
+func getConfigCheck(w http.ResponseWriter, _ *http.Request, ac autodiscovery.Component) {
 	var response response.ConfigCheckResponse
 
-	if common.AC == nil {
-		log.Errorf("Trying to use /config-check before the agent has been initialized.")
-		setJSONError(w, fmt.Errorf("agent not initialized"), 503)
-		return
-	}
-
-	configSlice := common.AC.LoadedConfigs()
+	configSlice := ac.LoadedConfigs()
 	sort.Slice(configSlice, func(i, j int) bool {
 		return configSlice[i].Name < configSlice[j].Name
 	})
 	response.Configs = configSlice
-	response.ResolveWarnings = autodiscovery.GetResolveWarnings()
-	response.ConfigErrors = autodiscovery.GetConfigErrors()
-	response.Unresolved = common.AC.GetUnresolvedTemplates()
+	response.ResolveWarnings = autodiscoveryimpl.GetResolveWarnings()
+	response.ConfigErrors = autodiscoveryimpl.GetConfigErrors()
```

Comment:
Similar question about having `GetResolveWarnings` and `GetConfigErrors` as part of the global implementation and not part of the component interface. 

Any reaosn why they are not part of the autodiscovery component interface?